### ISSUE TITLE
XMLWriter::startElement(): Invalid Element Name

### ISSUE THREAD

**Issue by notihnio** [braintree/braintree_php#184]
### General information

* SDK/Library version: 3.23.0
* Environment: Sandbox
* Language, language version, and OS: php 5.6 Mac OS 

### Issue description

I want to set a payment method as a default method for a customer

**My code** 

```
 $result = \Braintree_Customer::update($userId, [
                    ["defaultPaymentMethodToken" => $cardId ]
            ]);
```

**The error**

XMLWriter::startElement(): Invalid Element Name
File: vendor/braintree/braintree_php/lib/Braintree/Xml/Generator.php
Line:73

**Comment by sdcoffey** [comment]
Hey @notihnio,

Looks like your params hash is nested improperly. You have an outer hash around `["defaultPaymentMethodToken" => $cardId ]` that's likely the cause of the error. Please see [the docs](https://developers.braintreepayments.com/reference/request/customer/update/php) for updating a customer. 

tldr, your code should look like this:
```php
 $result = \Braintree_Customer::update($userId, 
                    ["defaultPaymentMethodToken" => $cardId ]
            );
```

 Let me know if this solves your issue.

**Comment by bluk** [comment]
Closing due to inactivity. If this still an issue, please leave a comment and we can reopen the issue. Thanks!

**Comment by mcoenca** [comment]
In case you have this kind of error and it's hard to debug, I was passing a badly ordered array of addon Ids due to an `array_filter` without `array_values`

```
BUGGY CODE
"add-ons"
array:1 [
  "add" => array:1 [
    1 => array:1 [
      "inherited-from-id" => "ambitieux_manche"
    ]
  ]
]
```

```
WORKING CODE
"add-ons"
array:1 [
  "add" => array:1 [
    0 => array:1 [
      "inherited-from-id" => "ambitieux_manche"
    ]
  ]
]
```
